File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums_dict = {}
        for num in nums:
            nums_dict[num] = nums_dict.get(num, 0) + 1
            if nums_dict[num] > len(nums)/2:
                return num
        
        

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        if (nums[0] < nums[nums.size()-1]) {
            return nums[0];
        }
        int l = 0, r = nums.size()-1, m = (nums.size() - 1)/2;
        while (l != m) {
            if (nums[m] > nums[l]) {
                l = m;
                m = (r+l)/2;
            } else {
                r = m;
                m = (r+l)/2;
            }
        }
        return nums[r];

    }
};
